src/VisualStudio.Roslyn.SDK/ComponentDebugger/LaunchSettingsProvider.cs

```diff
@@ -0,0 +1,101 @@
+    [AppliesTo(Constants.RoslynComponentCapability)]
+    public class LaunchSettingsProvider : ILaunchSettingsUIProvider
+    {
+        private readonly IProjectThreadingService _threadingService;
+        private readonly UnconfiguredProject _unconfiguredProject;
+        private readonly LaunchSettingsManager _launchSettingsManager;
+        private readonly DebuggerOptionsViewModel _viewModel;
+
+        private ImmutableArray<UnconfiguredProject> _projects;
+        private IWritableLaunchProfile? _launchProfile;
+
+        [ImportingConstructor]
+        [Obsolete("This exported object must be obtained through the MEF export provider.", error: true)]
+        public LaunchSettingsProvider(IProjectThreadingService threadingService, UnconfiguredProject unconfiguredProject, LaunchSettingsManager launchSettingsManager)
+        {
+            _threadingService = threadingService;
+            _unconfiguredProject = unconfiguredProject;
+            _launchSettingsManager = launchSettingsManager;
+            _viewModel = new DebuggerOptionsViewModel(IndexChanged);
+        }
+
+        public string CommandName { get => Constants.CommandName; }
+
+        // https://github.com/dotnet/roslyn-sdk/issues/730 : localization
+        public string FriendlyName { get => "Roslyn Component"; }
+
+        public UserControl? CustomUI { get => new DebuggerOptions() { DataContext = _viewModel }; }
+
+        public void ProfileSelected(IWritableLaunchSettings curSettings)
+        {
+            _launchProfile = curSettings?.ActiveProfile;
+            _threadingService.ExecuteSynchronously(UpdateViewModelAsync);
+        }
+
+        public bool ShouldEnableProperty(string propertyName)
+        {
+            // we disable all the default options for a debugger.
+            // in the future we might want to enable env vars and (potentially) the exe to allow
+            // customization of the compiler used?
+            return false;
+        }
+
+        private async Task UpdateViewModelAsync()
+        {
+            var targetProjects = ArrayBuilder<UnconfiguredProject>.GetInstance();
+
+            // get the output assembly for this project
+            var projectArgs = await _unconfiguredProject.GetCompilationArgumentsAsync().ConfigureAwait(false);
+            var targetArg = projectArgs.LastOrDefault(a => a.StartsWith("/out:", StringComparison.OrdinalIgnoreCase));
+            var target = Path.GetFileName(targetArg);
+
+            var projectService = _unconfiguredProject.Services.ProjectService;
+            foreach (var targetProjectUnconfigured in projectService.LoadedUnconfiguredProjects)
+            {
+                // check if the args contain the project as an analyzer ref
+                foreach (var arg in await targetProjectUnconfigured.GetCompilationArgumentsAsync().ConfigureAwait(false))
+                {
+                    if (arg.StartsWith("/analyzer", StringComparison.OrdinalIgnoreCase)
```

Comment:
```suggestion
                    if (arg.StartsWith("/analyzer:", StringComparison.OrdinalIgnoreCase)
```
To be consistent with the `StartsWith` above.